sympy/polys/densebasic.py

```diff
@@ -2080,54 +2080,69 @@ def dup_from_list(f: list[Er], K: Domain[Er]) -> dup[Er]:
     return dup_strip([K.convert(c) for c in f])
 
 
-def dup_print(f: dup[Er], sym: str) -> None:
+def dup_pretty(f: dup[Er], sym: str, *, ascending: bool = False) -> str:
     """
-    Print a polynomial in ``K[x]``.
+    Return a string representation of a polynomial in ``K[x]``.
 
     Examples
     ========
 
     >>> from sympy.polys.domains import ZZ
-    >>> from sympy.polys.densebasic import dup_print, dup_from_list
+    >>> from sympy.polys.densebasic import dup_pretty, dup_from_list
     >>> f = dup_from_list([1, 0, 5, 0, 7], ZZ)
-    >>> dup_print(f, 'x')
-    x**4 + 5*x**2 + 7
-
+    >>> dup_pretty(f, 'x')
+    'x**4 + 5*x**2 + 7'
+    >>> dup_pretty(f, 'x', ascending=True)
+    '7 + 5*x**2 + x**4'
     """
-    if isinstance(sym, tuple):
-        sym = sym[0]
 
     if not f:
-        print("0")
-        return
+        return "0"
 
     deg = dup_degree(f)
-    terms = []
-
-    for i, coeff in enumerate(f):
-        d = deg - i
 
+    def format_term(coeff, exp):
         if coeff == 0:
-            continue
+            return None
+        if exp == 0:
+            return str(coeff)
+        elif exp == 1:
+            base = sym
+        else:
+            base = f"{sym}**{exp}"
 
-        if d == 0:
-            term = f"{coeff}"
-        elif d == 1:
-            if coeff == 1:
-                term = f"{sym}"
-            elif coeff == -1:
-                term = f"-{sym}"
-            else:
-                term = f"{coeff}*{sym}"
+        if coeff == 1:
+            return base
+        elif coeff == -1:
+            return f"-{base}"
         else:
-            if coeff == 1:
-                term = f"{sym}**{d}"
-            elif coeff == -1:
-                term = f"-{sym}**{d}"
-            else:
-                term = f"{coeff}*{sym}**{d}"
+            return f"{coeff}*{base}"
+
+    terms = []
+    for i, coeff in enumerate(f):
+        t = format_term(coeff, deg - i)
+        if t is not None:
+            terms.append(t)
 
-        terms.append(term)
+    if ascending:
+        terms.reverse()
 
-    poly = " + ".join(terms).replace("+ -", "- ")
-    print(poly)
+    return " + ".join(terms).replace("+ -", "- ")
+
+
+def dup_print(f: dup[Er], sym: str, *, ascending: bool = False) -> None:
+    """
+    Print a polynomial in ``K[x]``.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.domains import ZZ
+    >>> from sympy.polys.densebasic import dup_print, dup_from_list
+    >>> f = dup_from_list([1, 0, 5, 0, 7], ZZ)
+    >>> dup_print(f, 'x')
+    x**4 + 5*x**2 + 7
+    >>> dup_print(f, 'x', ascending=True)
+    7 + 5*x**2 + x**4
+    """
```

Comment:
Are these functions in the docs?

I think it would be good to have See Also pointing to `dup_pretty` here and also in reverse from `dup_print` to here.

Reply:
I think it is reasonable.

A question that is not easy to answer right now without thinking about the bigger picture of all of the classes is whether it makes sense for the series ring itself to have a symbol like:
```python
R = power_series_ring(QQ, 'z')
```
I think maybe we don't want that in these low-level types because it would complicate things and it is better to have anonymous generators like the `dup` functions. The difference between this and the `dup` functions is just that you have a ring object so there is a place to store the name. Having the names complicates things though because then you need to keep track of that name everywhere when creating new rings or converting between them.